src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -793,17 +794,21 @@ export class SearchSource {
             const field2Name = this.getFieldName(fld2);
             return field1Name === field2Name;
           }
-        ).map((fld: SearchFieldValue) => {
-          const fieldName = this.getFieldName(fld);
-          if (Object.keys(docvaluesIndex).includes(fieldName)) {
-            // either provide the field object from computed docvalues,
-            // or merge the user-provided field with the one in docvalues
-            return typeof fld === 'string'
-              ? docvaluesIndex[fld]
-              : this.getFieldFromDocValueFieldsOrIndexPattern(docvaluesIndex, fld, index);
-          }
-          return fld;
-        });
+        )
+          .filter((fld: SearchFieldValue) => {
+            return !metaFields.includes(this.getFieldName(fld));
+          })
```

Comment:
IMO it would make it a bit more readable if we moved this up to L718 where we are originally applying the wildcard filters. That way we are handling all of the source filtering in the same place, and this section is only used for dealing with copying the docvalues.

Reply:
I think in that line we still don't have list of fields, but `fields: '*'`. Waiting for kibana to boot and I'll know for sure.

Reply:
Just checked, at line L718 it's still `fields: '*'`, before we apply the logic to get the actual fields.

Reply:
Ah okay, I missed that then. Thanks for confirming!